### PR TITLE
`Face.family` as a vec

### LINKED ISSUE #1: Support localized family names

### PR DESCRIPTION
**Pull request by declantsien**
Currently, fontdb has issues matching fonts with multi (localized) family names.
This merge request store all family name under `Face.family`, and iter over them when `query`.

This breaks the existing `Face` interface.

Probably fixes #1?

### CODE REVIEW

**Comment by RazrFalcon** [comment]
There is definitely a problem, but this is sadly not a solution. 

The issue here is that I don't know what is the correct output to begin with. Someone has to go thought every OS library (win, mac, linux) and check what they do so we could match them.
Storing all names will not help at all, potentially even breaking matching. And we have to at least preserve language tags, otherwise those names are pointless.

Basically, if it was that easy, it had been implemented already.

**Comment by declantsien** [comment]
Oh, No problem, closing.

In my case, I need "等距更紗黑體 CL" and "Sarasa Mono CL" to match
the same font. Language tags are not used in my application.

Currently `fontdb` matches "等距更紗黑體 CL" but not "Sarasa Mono CL",
since name "Sarasa Mono CL" is simply dropped when initialize the data
store.

I have tried https://github.com/matprec/rust-font-loader.git which using
native framework to match fonts. But it seems to cause memory leaks last
time I checked with Xcode Instruments. Also `rust-font-loader` only
matchs "Sarasa Mono CL" but not "等距更紗黑體 CL"

> Someone has to go thought every OS library (win, mac, linux) and check
> what they do so we could match them.

Until then :)

**Comment by RazrFalcon** [comment]
Can you provide the font you're using? Or a link to it. I want to look into it.

**Comment by declantsien** [comment]
I am using Sarasa-Gothic.
Here's the link.
https://github.com/be5invis/Sarasa-Gothic/releases/tag/v0.38.0

**Comment by RazrFalcon** [comment]
Interesting. First thing we should do is actually #32

As for ability to choose both, I will have to take a look. Even for a basic `Vec<T>` solution we have to figure out what to use as a language tag and how to make it user-friendly. Basically a design and not implementation problem.

**Comment by declantsien** [comment]
> First thing we should do is actually #32

I also need to prefer English family_name on macOS. I just use
`family.is_ascii()`, not sure if that is the right way or not.

> As for ability to choose both, I will have to take a look. Even for a
> basic `Vec<T>` solution we have to figure out what to use as a
> language tag and how to make it user-friendly. Basically a design and
> not implementation problem.

Yeah, totally agree. Looking forward to it.

**Comment by RazrFalcon** [comment]
Should be working just fine now. You can use both English and localized family names.

**Comment by declantsien** [comment]
Yevhenii Reizner ***@***.***> writes:

> Should be working just fine now. You can use both English and localized family names.

Nice, its working. Just deleted my fork.
